pandas/core/indexes/base.py

```diff
@@ -1710,6 +1710,42 @@ def _invalid_indexer(self, form, key):
                             kind=type(key)))
 
     def get_duplicates(self):
+        """
+        Extract duplicated index elements.
+
+        This function returns a sorted list of index elements which appear more
+        than once in the index.
+
+        Returns
+        -------
+        array-like
+            List of duplicated indices.
+
+        See Also
+        --------
+        :meth:`Index.duplicated` : Return boolean array denoting duplicates.
+        :meth:`Index.drop_duplicates` : Return Index with duplicates removed.
+
+        Examples
+        --------
+        >>> pd.Index([1, 2, 3, 4]).get_duplicates()
+        []
+        >>> pd.Index([1, 2, 2, 3, 3, 3, 4]).get_duplicates()
+        [2, 3]
+        >>> pd.Index([1, 2, 3, 2, 3, 4, 3]).get_duplicates()
+        [2, 3]
+        >>> pd.Index(['a', 'b', 'b', 'c', 'c', 'c', 'd']).get_duplicates()
+        ['b', 'c']
+        >>> dates = pd.to_datetime(['2018-01-01', '2018-01-02',
+        ...                         '2018-01-03', '2018-01-03'],
+        ...                        format='%Y-%m-%d')
+        >>> pd.Index(pd.to_datetime(dates, format='%Y-%m-%d')).get_duplicates()
```

Comment:
you don't need to repeat the `to_datetime` here. 

Can you also add a sentence here before the example explaining that for DatetimeIndex, it does not return a list but a DatetetimeIndex ?